### PR TITLE
FHIRNotificationService should emit event for delete #983

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-notification/src/main/java/com/ibm/fhir/notification/FHIRNotificationService.java [hunk 1]
```diff
@@ -147,13 +148,10 @@ public void afterUpdate(FHIRPersistenceEvent pEvent) throws FHIRPersistenceInter
...
-    public void beforeCreate(FHIRPersistenceEvent pEvent) throws FHIRPersistenceInterceptorException {
-        // Nothing to do for 'beforeCreate'.
-    }
-
-    @Override
-    public void beforeUpdate(FHIRPersistenceEvent event) throws FHIRPersistenceInterceptorException {
-        // Nothing to do for 'beforeUpdate'.
+    public void afterDelete(FHIRPersistenceEvent pEvent) throws FHIRPersistenceInterceptorException {
+        if (shouldPublish(pEvent)) {
+            this.publish(buildNotificationEvent("delete", pEvent));
```
Comment (lmsurpre) [inline fhir-notification/src/main/java/com/ibm/fhir/notification/FHIRNotificationService.java:153]: that was easy :-)

**Comment by lmsurpre** [comment]
Question:  do we have any e2e / integration tests for the audit?

**Comment by prb112** [comment]
> Question: do we have any e2e / integration tests for the audit?

Yes for Audit, no for Notifications.

**Comment by lmsurpre** [comment]
> Yes for Audit, no for Notifications.
Haha, yeah thats what I meant (you read my mind).  We used to have websocket tests which probably exercised some of the notification infrastructure.
We also did some work to spin up a NATS.io cluster as part of our docker-compose for the db2 integration tests, but I agree it looks like we never took the final step to automate that:
> Note: If you are testing NATS notifications, invoke the NATS subscriber via `node fhir-server-test/src/test/nodejs/nats-subscriber`.  If this is your first time, install the dependencies first by installing [Node.js](https://nodejs.org/en/download) (if not already installed) and running `(cd fhir-server-test/src/test/nodejs && npm install)`.

I've opened https://github.com/IBM/FHIR/issues/1936 for getting that added.

**Review by lmsurpre — APPROVED** [review]
LGTM

**Comment by prb112** [comment]
> > Yes for Audit, no for Notifications.
> > Haha, yeah thats what I meant (you read my mind).  We used to have websocket tests which probably exercised some of the notification infrastructure.
> > We also did some work to spin up a NATS.io cluster as part of our docker-compose for the db2 integration tests, but I agree it looks like we never took the final step to automate that:
> > Note: If you are testing NATS notifications, invoke the NATS subscriber via `node fhir-server-test/src/test/nodejs/nats-subscriber`.  If this is your first time, install the dependencies first by installing [Node.js](https://nodejs.org/en/download) (if not already installed) and running `(cd fhir-server-test/src/test/nodejs && npm install)`.
> 
> I've opened #1936 for getting that added.

It shouldn't be too hard to add.

**Pull request merged**